superset/assets/javascripts/SqlLab/components/SaveQuery.jsx

```diff
@@ -0,0 +1,139 @@
+/* global notify */
+import React from 'react';
+import { FormControl, FormGroup, Overlay, Popover, Row, Col } from 'react-bootstrap';
+import Button from '../../components/Button';
+const $ = window.$ = require('jquery');
+
+const propTypes = {
+  defaultLabel: React.PropTypes.string,
+  sql: React.PropTypes.string,
+  schema: React.PropTypes.string,
+  dbId: React.PropTypes.number,
+  animation: React.PropTypes.bool,
+};
+const defaultProps = {
+  defaultLabel: 'Undefined',
+  animation: true,
+};
+
+class SaveQuery extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      description: '',
+      label: props.defaultLabel,
+      showSave: false,
+    };
+    this.toggleSave = this.toggleSave.bind(this);
+    this.onSave = this.onSave.bind(this);
+    this.onCancel = this.onCancel.bind(this);
+    this.onLabelChange = this.onLabelChange.bind(this);
+    this.onDescriptionChange = this.onDescriptionChange.bind(this);
+  }
+  onSave() {
+    const url = '/savedqueryviewapi/api/create';
```

Comment:
we should put this request in `actions.js` https://github.com/ascott/superset/blob/master/superset/assets/javascripts/SqlLab/actions.js